### PR TITLE
Fix for error code 50

### PR DESCRIPTION
**Pull request by wingZero21**
### Cookbook version

4.1.9

### Chef-client version

12.10.24

### Platform Details

Windows Server 2012

### Scenario:

Attempting to install .net 35 pre-requisites

### Steps to Reproduce:

Running recipe iis::mod_aspnet

### Expected Result:

When running I am getting the below errors.
```
==> default: [2016-06-28T01:06:44-07:00] FATAL: Mixlib::ShellOut::ShellCommandFailed: windows_feature[NetFx3] (iis::mod_aspnet line 31) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0, 42, 127, 3010], but received '50'
==> default: ---- Begin output of C:\Windows\system32\dism.exe /online /enable-feature /featurename:NetFx3 /norestart   ----
==> default: STDOUT: Deployment Image Servicing and Management tool
==> default:
==> default: Version: 6.3.9600.17031
==> default:
==> default:
==> default:
==> default: Image Version: 6.3.9600.17031
==> default:
==> default:
==> default:
==> default: Enabling feature(s)
==> default:
==> default:
==> default:
==> default: Error: 50
==> default:
==> default:
==> default:
==> default: The operation is complete but NetFx3 feature was not enabled.
==> default:
==> default: A required parent feature may not be enabled. You can use the /enable-feature /all option to automatically enable each parent feature from the following list. If the parent feature(s) are already enabled, refer to the log file for further diagnostics.
==> default:
==> default: NetFx3ServerFeatures
==> default:
==> default: The DISM log file can be found at C:\Windows\Logs\DISM\dism.log
==> default: STDERR:
==> default: ---- End output of C:\Windows\system32\dism.exe /online /enable-feature /featurename:NetFx3 /norestart   ----
==> default: Ran C:\Windows\system32\dism.exe /online /enable-feature /featurename:NetFx3 /norestart   returned 50
```
### Potential Fix

Amended the mod_aspnet recipe detailed below.

Details:
Original
```
include_recipe 'iis'
include_recipe 'iis::mod_isapi'

features = if Opscode::IIS::Helper.older_than_windows2008r2?
             %w(NET-Framework)
           else
             %w(IIS-NetFxExtensibility IIS-ASPNET)
           end

features.each do |feature|
  windows_feature feature do
    action :install
  end
end
```
Amendment
```
include_recipe 'iis'
include_recipe 'iis::mod_isapi'

features = if Opscode::IIS::Helper.older_than_windows2008r2?
             %w(NET-Framework)
           else
             %w(NetFx3 IIS-NetFxExtensibility IIS-ASPNET)
           end

features.each do |feature|
  windows_feature feature do
    action :install
    all true
  end
end
```

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @wingZero21

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by chef-supermarket** [comment]
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

File: recipes/mod_aspnet.rb [hunk 1]
```diff
@@ -24,11 +24,12 @@
 features = if Opscode::IIS::Helper.older_than_windows2008r2?
              %w(NET-Framework)
            else
-             %w(IIS-NetFxExtensibility IIS-ASPNET)
+             %w(NetFx3 IIS-NetFxExtensibility IIS-ASPNET)
```
Comment (EasyAsABC123) [inline recipes/mod_aspnet.rb]: Un-needed with `all true`

File: recipes/mod_aspnet.rb [hunk 2]
```diff
@@ -24,11 +24,12 @@
...
            end
 
 features.each do |feature|
   windows_feature feature do
     action :install
+    all true
```
Comment (EasyAsABC123) [inline recipes/mod_aspnet.rb]: Should just be this

**Comment by EasyAsABC123** [comment]
Please validate that the PR resolves the issue.  This is basically stating that .NET 3 needs to be installed, there are multiple cookbooks already out there that will install .net 3.

**Comment by wingZero21** [comment]
Hi,

Yes this fixed the issue for me.

Cheers

**Comment by EasyAsABC123** [comment]
@wingZero21 I meant the code changes that I requested, where you only do one or the other fix.

**Comment by EasyAsABC123** [comment]
Pushed to supermarket, v4.1.10

**Pull request merged**